src/core/toga/fonts.py

```diff
@@ -34,3 +36,95 @@ def bind(self, factory):
+        )
+        _REGISTERED_FONT_CACHE[registered_font.key] = registered_font
+
+    @staticmethod
+    def find_registered_font(family, weight=NORMAL, style=NORMAL, variant=NORMAL):
+        """
+        Returns the RegisteredFont which has been registered with Font.register()
+        When the exact family/weight/style/variant combination has not been registered, the method tries
+        to return the RegisteredFont for the registered family only, e.g.
+
+        registered_font = Font.find_registered_font("Roboto", weight=Font.BOLD)
+
+        Args:
+            family (str):  The font family name
+            weight (str):  The font weight: Font.NORMAL (default) or a value from Font.FONT_WEIGHTS
+            style (str):   The font style: Font.NORMAL (default) or a value from Font.FONT_STYLES
+            variant (str): The font variant: Font.NORMAL (default) or a value from Font.FONT_VARIANTS
+        Returns:
+            The RegisteredFont or None
+        """
+        font_key = RegisteredFont.make_key(family, weight, style, variant)
+        try:
+            return _REGISTERED_FONT_CACHE[font_key]
+        except KeyError:
+            try:
+                if weight != NORMAL or style != NORMAL or variant != NORMAL:
+                    font_key_family = RegisteredFont.make_key(
+                        family, NORMAL, NORMAL, NORMAL
+                    )
+                    print(
+                        "Registered font "
+                        + font_key
+                        + " not found. Using "
+                        + font_key_family
+                        + " instead."
+                    )
+                    return _REGISTERED_FONT_CACHE[font_key_family]
+                else:
+                    return None
+            except KeyError:
+                print("Registered font " + font_key_family + " not found.")
+                return None
+
+
+class RegisteredFont:
+    def __init__(self, family, path, weight=NORMAL, style=NORMAL, variant=NORMAL):
+        self.family = family
+        self.path = path
+        self.weight = weight if weight in constants.FONT_WEIGHTS else NORMAL
+        self.style = style if style in constants.FONT_STYLES else NORMAL
+        self.variant = variant if variant in constants.FONT_VARIANTS else NORMAL
+
+    @property
+    def key(self):
+        return "<RegisteredFont family={} weight={} style={} variant={}>".format(
+            self.family, self.weight, self.style, self.variant
+        )
+
+    @staticmethod
+    def make_key(family, weight, style, variant):
+        if weight not in constants.FONT_WEIGHTS:
+            weight = NORMAL
+        if style not in constants.FONT_STYLES:
+            style = NORMAL
+        if variant not in constants.FONT_VARIANTS:
+            variant = NORMAL
+        return "<RegisteredFont family={} weight={} style={} variant={}>".format(
+            family, weight, style, variant
```

Comment:
This effectively duplicates the key construction logic. This should be a single utility function, rather than a duplication.

Reply:
OK, I removed the RegisteredFont class